Fix accessibility issues

solita-antti-mottonen opened the pull request:
The colours in tailwind config aren't the official styleguide colors, as
the site pre-dates the styleguide. Added some of the official colors to
the configuration.

Background color was not set for the whole page. Footer had one gray
shade, while the rest of the page was transparent and thus used the
system default, which incidentally was the same shade as footer at least
for me. Explicitly set the background color for the whole application to
be the official gray shade. It differs slightly from the color used
elsewhere in the site, but not enought to be noticable at least by me.

Links in the footer did not have enough contrast. Added an option to
the Link component to add additional classes, and switched the footer
links to use the darker green meant for small text on non-white
background.

Change history-page in user history had a column for public information.
Public information was in bold green, whilst, non-public information was
in light grey, which didn't have enough contrast on gray rows. Changed the
column to show only the public fields in the same style as all other
information in the table.

solita-antti-mottonen committed:
Fix accessibility issues

The colours in tailwind config aren't the official styleguide colors, as
the site pre-dates the styleguide. Added some of the official colors to
the configuration.

Background color was not set for the whole page. Footer had one gray
shade, while the rest of the page was transparent and thus used the
system default, which incidentally was the same shade as footer at least
for me. Explicitly set the background color for the whole application to
be the official gray shade. It differs slightly from the color used
elsewhere in the site, but not enought to be noticable at least by me.

Links in the footer did not have enough contrast. Added an option to
the Link component to add additional classes, and switched the footer
links to use the darker green meant for small text on non-white
background.

Change history-page in user history had a column for public information.
Public information was in bold green, whilst, non-public information was
in light grey, which didn't have enough contrast on gray rows. Changed the
column to show only the public fields in the same style as all other
information in the table.

Juholei commented on src/components/Link/Link.svelte:
```diff
@@ -26,9 +27,9 @@
 <a {href} {target} class:disabled class:font-bold={bold} on:click>
   {#each Maybe.toArray(icon) as i}
-    <span class="font-icon mr-1">{i}</span>
+    <span class="font-icon mr-1}">{i}</span>
```
Tän ei varmaan kuuluisi olla muuttunut?

solita-antti-mottonen replied:
Ei

Juholei commented:
Ihan fiksulta näyttää koodin puolesta, en tosin katsonut että miten ulkonäkö muuttuu värien myötä.

solita-antti-mottonen commented:
Vanhat ja uudet versiot muutoksista. Sininen alatunnisteessa on meillä määrittelemätön (läpinäkyvä) väri ja riippuu siis selaimen asetuksista
<img width="1589" alt="alatunniste-vanha" src="https://github.com/solita/etp-front/assets/125372602/a03de818-d886-4dce-8923-e8c8028715de">
<img width="1658" alt="alatunniste-uusi" src="https://github.com/solita/etp-front/assets/125372602/9ce33fd5-3f01-4ad7-aba6-9603c2217fd1">

<img width="197" alt="et-tallenna-vanha" src="https://github.com/solita/etp-front/assets/125372602/b0eb9526-64d7-4e7d-bad2-d3492be19b6f">
<img width="179" alt="et-tallenna-uusi" src="https://github.com/solita/etp-front/assets/125372602/03a0a41f-05ab-4a2a-802f-3877abe91909">

<img width="510" alt="muutoshistoria-vanha" src="https://github.com/solita/etp-front/assets/125372602/f877ba37-f0ac-4b56-9eed-9982baa3786c">
<img width="507" alt="muutoshistoria-uusi" src="https://github.com/solita/etp-front/assets/125372602/809ba012-a79e-4526-b8fa-ded1a7e2ac98">

solita-antti-mottonen commented:
Harmaan sävyt lähekkäin
![harmaasävyjen erot](https://github.com/solita/etp-front/assets/125372602/47902eaf-b27f-4f8b-b741-46905e388396)

<img width="1589" alt="alatunniste-vanha" src="https://github.com/solita/etp-front/assets/125372602/3b4cfd31-86e3-4cdc-8f5b-13a6c9198cd4">

![alatunniste-uusi](https://github.com/solita/etp-front/assets/125372602/ba535ed9-b88c-4723-92d4-aeb9cb5af12a)



<img width="510" alt="muutoshistoria-vanha" src="https://github.com/solita/etp-front/assets/125372602/6e6ee6c8-4043-420d-acd2-86e75ae1505b">

![muutoshistoria-uusi](https://github.com/solita/etp-front/assets/125372602/4049fc68-2838-4e76-980f-2addb461df99)


<img width="197" alt="et-tallenna-vanha" src="https://github.com/solita/etp-front/assets/125372602/e89c5cd7-5994-4d0f-9e15-5da3c0765a0d">

![et-tallenna-uusi](https://github.com/solita/etp-front/assets/125372602/265aab8a-908b-4ba8-bff8-9107f161a7f8)

solita-antti-mottonen commented:
Viesteissä ainut matala kontrasti jonka löysin on minä linkki. Teksti on lihavoitu, eli menee speksiin. 

Omissa tiedoissa on disabloituja elementtejä, joiden kontrasti on huono. Nämäkin ovat kuitenkin speksin mukaan ok.

solita-antti-mottonen commented on src/pages/footer/Footer.svelte:
```diff
@@ -21,40 +21,48 @@
     </div>
     <div class="flex flex-col space-y-2">
-      <span class="uppercase font-bold text-secondary">
+      <span class="uppercase font-bold text-ara-green-small-text">
```
Poista tämä

solita-antti-mottonen committed:
Switch to white background

White seems to be more used default setting. Switch to white background
instead of gray.
